script/v2/manual-runs/scWETHv2Rebalance.s.sol

```diff
@@ -0,0 +1,214 @@
+
+            totalFlashLoanAmount += flashLoanAmounts[i];
+            totalAllocationPercent += allocationPercents[i];
+        }
+
+        require(totalAllocationPercent == 1e18, "totalAllocationPercent != 100%");
+
+        bytes[] memory callData = new bytes[](n+1);
+
+        uint256 wethSwapAmount = _amount + totalFlashLoanAmount;
+        bytes memory swapData = demoSwapData.length != 0 ? demoSwapData : getSwapDataWethToWstEth(wethSwapAmount);
+
+        callData[0] =
+            abi.encodeWithSelector(BaseV2Vault.zeroExSwap.selector, C.WETH, C.WSTETH, wethSwapAmount, swapData, 0);
+
+        for (uint256 i = 1; i < n + 1; i++) {
+            callData[i] = abi.encodeWithSelector(
+                scWETHv2.supplyAndBorrow.selector, adapters[i - 1].id(), supplyAmounts[i - 1], flashLoanAmounts[i - 1]
+            );
+        }
+
+        return (callData, totalFlashLoanAmount);
+    }
+
+    function getSwapDataWethToWstEth(uint256 _amount) public returns (bytes memory swapData) {
+        string memory url = string(
+            abi.encodePacked(
+                "https://api.0x.org/swap/v1/quote?buyToken=0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0&sellToken=WETH&sellAmount=",
+                _amount.toString()
+            )
+        );
+
+        string[] memory headers = new string[](1);
+        headers[0] = string(abi.encodePacked("0x-api-key: ", vm.envString("ZEROX_API_KEY")));
+        (uint256 status, bytes memory data) = url.get(headers);
+
+        require(status == 200, "0x GET request Failed");
+
+        string memory json = string(data);
+
+        swapData = json.readBytes(".data");
+    }
+
+    function _calcSupplyBorrowFlashLoanAmount(IAdapter _adapter, uint256 _amount)
+        internal
+        view
+        returns (uint256 flashLoanAmount)
+    {
+        uint256 debt = vault.getDebt(_adapter.id());
+        uint256 collateral = getCollateralInWeth(_adapter);
+
+        uint256 target = targetLtv[_adapter].mulWadDown(_amount + collateral);
+
+        // calculate the flashloan amount needed
+        flashLoanAmount = (target - debt).divWadDown(C.ONE - targetLtv[_adapter]);
+    }
+
+    function _logs() internal view {
+        uint256 collateralInWeth = priceConverter.wstEthToEth(vault.totalCollateral());
+        uint256 debt = vault.totalDebt();
+        console2.log("\n Total Collateral %s weth", collateralInWeth);
+        console2.log("\n Total Debt %s weth", debt);
+        console2.log("\n Invested Amount %s weth", collateralInWeth - debt);
+        console2.log("\n Total Assets %s weth", vault.totalAssets());
+        console2.log("\n Net Leverage", getLeverage());
+        console2.log("\n Net LTV", debt.divWadUp(collateralInWeth));
+    }
+
+    ///////////////////////////////// FOR TESTING ONLY ///////////////////////////
+    bytes demoSwapData;
+
+    function setDemoSwapData(bytes memory _data) external {
+        demoSwapData = _data;
+    }
```

Comment:
you don't actually need this, a better way would be to declare the `getSwapDataWethToWstEth` as virtual, override it in a harness contract used in tests and provide the swap data there.